[IMP] inventory: split removal strategy doc into 5 (#7223)
odoo/documentation: content/applications/inventory_and_mrp/inventory/warehouses_storage/advanced_operations_warehouse/removal.rst

--> Settings`. Under the :guilabel:`Traceability` heading, check the box beside :guilabel:`Lots &
Serial Numbers` to enable the feature.

.. image:: removal/enable-lots.png
   :align: center
   :alt: Enable lots and serial numbers.

Next, ensure the intended product is tracked by lots or serial numbers by navigating to the product
form through :menuselection:`Inventory --> Products --> Products` and selecting the desired product.
On the product form, switch to the :guilabel:`Inventory` tab, and under the :guilabel:`Tracking`
field, select either the :guilabel:`By Unique Serial Number` or :guilabel:`By Lots` options.

Expiration date
~~~~~~~~~~~~~~~

Enable the **expiration date** to track expiration dates, best before dates, removal dates, and
alert dates on a lot or serial number. This feature is only available with lots and serial numbers
enabled.

.. image:: removal/enable-expiration.png
   :align: center
   :alt: Enable expiration dates feature for FEFO.

Locations and routes
~~~~~~~~~~~~~~~~~~~~

**Storage locations** and **multi-step routes** are necessary features for setting **all** types of

> removal strategies on a location. However, these features are specifically required for the closest
> location removal strategy since it is only applied at the location level.

Author: ```suggestion
removal strategies on a location. 
```

removal strategy.

.. image:: removal/enable-pack.png
   :align: center
   :alt: Enable the packages feature.

.. seealso::
   - :ref:`Packages <inventory/management/packages>`
   - :doc:`2-step delivery <../../shipping_receiving/daily_operations/receipts_delivery_two_steps>`
   - :doc:`3-step delivery <../../shipping_receiving/daily_operations/delivery_three_steps>`

.. _inventory/removal/fifo:

First in, first out (FIFO)
==========================

The *first in, first out* (FIFO) removal strategy selects products with the earliest arrival dates.
This method is useful for companies selling products with short demand cycles, like clothes, to
prevent prolonged stock retention of specific styles.

.. example::
   Various quantities of the product, `T-shirt`, tracked by lot numbers, arrive on August 1st and
   August 25th. For an order made on September 1st, the :abbr:`FIFO (First In, First Out)` removal
   strategy  prioritizes lots that have been in stock the longest. So, products received on August
   1st are selected first for picking.

   .. image:: removal/fifo-example.png
      :align: center
      :alt: Illustration of FIFO selecting the oldest products in stock.

> To use the :abbr:`FIFO (First In, First Out)` removal strategy, enable the :guilabel:`Lots & Serial
> Numbers` feature and activate :guilabel:`Tracking` by lots or serial numbers on the product form.

Author: ```suggestion

```

> To use the :abbr:`FIFO (First In, First Out)` removal strategy, enable the :guilabel:`Lots & Serial
> Numbers` feature and activate :guilabel:`Tracking` by lots or serial numbers on the product form.

Author: ```suggestion

```

==================
Removal strategies
==================

For companies with warehouses, *removal strategies* determine **which** products are taken from the
warehouse, and **when**. For example, for perishable products, prioritizing those with the closest
expiration date is crucial to minimizing spoilage.

Leverage removal strategies to have Odoo automatically select how products are selected for orders.

The following are the removal strategies available in Odoo, detailing how the pickings are
determined, and the picking order:

.. list-table::
   :header-rows: 1
   :stub-columns: 1

   * -
     - FIFO
     - LIFO
     - FEFO
     - Closest Location
     - Least Packages
   * - Based on
     - Incoming date
     - Incoming date
     - Removal date
     - Location sequence
     - Package quantity
   * - Selection order
     - First in
     - Last in
     - First to expire
     - Alphanumeric name of location
     - Quantity closest to fulfilling demand

For comprehensive examples for how to use each removal strategy, refer to each individual
documentation page.

> .. seealso::
>    -

Author: ```suggestion

```

> .. _inventory/routes/strategies/set:

Reviewer: Hey @Felicious, I agree with @StraubCreative here. 

The custom anchor should align with the folder structure of the doc and the section you are anchoring in, so for this anchor, I would say either:
```
.. _inventory/warehouses_storage/removalConfig:
```
or
```
.. _inventory/warehouses_storage/removal-config:
```

> .. _inventory/routes/strategies/set:

Author: ```suggestion
.. _inventory/warehouses_storage/removal-config:
```

Configuration
=============

Removal strategies are set on either the product category or storage location.

Configure removal strategies on the location by going to :menuselection:`Inventory --> Configuration
--> Locations`, and selecting the desired location. On the location form, choose a removal strategy
from the :guilabel:`Removal Strategy` field's drop-down menu options.

.. important::
   To set a removal strategy on a location, the :guilabel:`Storage Locations` and
   :guilabel:`Multi-Step Routes` settings **must** be enabled in :menuselection:`Inventory -->
   Configuration --> Settings`.

   These features are **only** necessary when setting the removal strategy on a location.

Configure removal strategies on product categories by going to :menuselection:`Inventory -->
Configuration --> Product Categories` and selecting the intended product category. Next, choose a
removal strategy from the :guilabel:`Force Removal Strategy` drop-down menu options.

> .. admonition:: Priority
> 
>    When there are different removal strategies applied on the location and product category for a
>    product, the *force* removal strategy set on the :guilabel:`Product Category` is applied.

Reviewer 2: Don't know if I'd do a custom admonition here when an **Important** one seems to fit nicely here.

As well, I'd be *very* specific in this section in order to minimize misunderstandings.

```suggestion
.. important::
When there are different removal strategies applied on both the location and product category for a
product, the value set on the :guilabel:`Force Removal Strategy` field on the :guilabel:`Product
Category` form is applied as top priority.
```

   :stub-columns: 1

   * -
     - FIFO
     - LIFO
     - FEFO
     - Closest Location
     - Least Packages
   * - Required features
     - Lots & Serial Numbers
     - Lots & Serial Numbers
     - Lots & Serial Numbers, Expiration Date
     - Storage Locations, Multi-Step Routes
     - Packages

.. _inventory/removal/lots-setup:

Lots and serial numbers
~~~~~~~~~~~~~~~~~~~~~~~

Lots and serial numbers differentiate identical products and track information like arrival or
expiration dates. To enable this feature, navigate to :menuselection:`Inventory --> Configuration
--> Settings`. Under the :guilabel:`Traceability` heading, check the box beside :guilabel:`Lots &
Serial Numbers` to enable the feature.

.. image:: removal/enable-lots.png
   :align: center
   :alt: Enable lots and serial numbers.

Next, ensure the intended product is tracked by lots or serial numbers by navigating to the product
form through :menuselection:`Inventory --> Products --> Products`, and selecting the desired
product. On the product form, switch to the :guilabel:`Inventory` tab, and under the
:guilabel:`Tracking` field, select either the :guilabel:`By Unique Serial Number` or :guilabel:`By

> Lots` options.

Reviewer 2: After selecting this on the product form, the next step is to add the lot/serial # to the product so Odoo can actually track it properly. It's not enough to just enable the setting, so this last step is what completes the instruction. 

Therefore, I'd briefly list the step(s) on how to do this (e.g. **☰ Lots/Serial Numbers** smart button on product form) or point to other documentation that has that information.

Locations and routes
~~~~~~~~~~~~~~~~~~~~

**Storage locations** and **multi-step routes** are necessary features for setting **all** types of
removal strategies on a location. However, these features are specifically required for the closest
location removal strategy since it is only applied at the location level.

To activate these features, navigate to :menuselection:`Inventory --> Configuration --> Settings`.
Under the :guilabel:`Warehouse` heading, enable the :guilabel:`Storage Location` and
:guilabel:`Multi-Step Routes` features.

.. image:: removal/enable-location.png
   :align: center
   :alt: Enable the locations and route features.

Expiration date

> ~~~~~~~~~~~~~~~

Reviewer 2: There are no specific navigational steps here for this setting.

--> Locations`, and selecting the desired location. On the location form, choose a removal strategy
from the :guilabel:`Removal Strategy` field's drop-down menu options.

.. important::
   To set a removal strategy on a location, the :guilabel:`Storage Locations` and
   :guilabel:`Multi-Step Routes` settings **must** be enabled in :menuselection:`Inventory -->
   Configuration --> Settings`.

   These features are **only** necessary when setting the removal strategy on a location.

Configure removal strategies on product categories by going to :menuselection:`Inventory -->
Configuration --> Product Categories` and selecting the intended product category. Next, choose a
removal strategy from the :guilabel:`Force Removal Strategy` drop-down menu options.

.. important::
   When there are different removal strategies applied on both the location and product category for
   a product, the value set on the :guilabel:`Force Removal Strategy` field set on a
   :guilabel:`Product Category` form is applied as top priority.

   .. image:: removal/navigate-location-category.png
      :align: center

>       :alt: Change the Force Removal Strategy for either the Product Categories or Locations.

Reviewer 2: Was this image always set in the admonition like this or did it move since last review?
Seems to fit better right after line 44, up to you 🙂

for six cartons of eggs, by either going to the :menuselection:`Sales app` and create a new
quotation, or from the delivery orders dashboard in :menuselection:`Inventory --> Operations -->
Deliveries`. :guilabel:`Confirm` the sales order or click :guilabel:`Mark as Todo` on the draft
delivery order.

Doing so creates the delivery order for today, December 29th, and the lot numbers with the soonest
expiration dates are reserved thanks to the :abbr:`FEFO (First Expired, First Out)` removal
strategy.

To view the detailed pickings, click the :guilabel:`⦙≣ (bulleted list)` icon on the far right of the
egg's product line in the :guilabel:`Operations` tab of the delivery order. Doing so opens
the :guilabel:`Stock move` pop-up window.

In the :guilabel:`Stock move` window, the :guilabel:`Pick from` field details where the quantities
to fulfill the :guilabel:`Demand` are picked from. Since the order demanded six cartons of eggs,
using the :abbr:`FEFO (First Expired, First Out)` removal strategy, all five cartons from `LOT1`
with the removal date of February 26th are picked. The remaining carton is selected from the `LOT2`,

> which has a removal date of May 11.

Author: Thanks for noticing that! I'll make the change (:

   .. image:: removal/locations.png
      :align: center
      :alt: Show a mockup of real storage location in a warehouse.

   The sublocation `Pallet` is on the ground level, so products stored here are easier to retrieve,
   compared to requiring a forklift to reach `Rack 1` and `Rack 2`. The storage locations were
   strategically named in alphabetic order based on ease of access.

.. important::
   To use this removal strategy, the :guilabel:`Storage Locations` and :guilabel:`Multi-Step Routes`
   settings **must** be enabled in :menuselection:`Inventory --> Configuration --> Settings`.

Location names
--------------

Create new or rename locations by navigating to :menuselection:`Inventory --> Configuration -->
Locations`. Select the desired location and type the :guilabel:`Location Name`.

Once the locations are named in alphabetical order based on the location's proximity from the output
or packing location, set the removal strategy on the parent location. To do that, in the
:guilabel:`Locations` list, select the parent location of the alphabetically named storage
locations.

Doing so opens the form for the parent location. In the :guilabel:`Removal Strategy` field, select

> :guilabel:`Closest Location`.

Author: added an internal link to a doc that explains this!

:menuselection:`Inventory --> Configuration --> Settings`. Then, check the box beside
:guilabel:`Packages` to enable the feature.

Then, set the removal strategy on the :ref:`product category or storage location
<inventory/routes/strategies/set>`.

To see how the least packages removal strategy works, consider the product, `Flour`, in the
following example. The product's :guilabel:`Units of Measure` field on the product form set to `kg`.
The product is stored in packages of one hundred kilograms, with one package with `54 kg` remaining.

.. tip::
   To check the product's on-hand stock, navigate to the product form and click the :guilabel:`On
   Hand` smart button.

   .. image:: removal/on-hand-flour.png
      :align: center
      :alt: Show on-hand stock in each package.

Navigate to the product category, `Wheats`, by going to :menuselection:`Inventory --> Configuration
--> Product Categories` and selecting the desired category. On the product category form, set the
:guilabel:`Force Removal Strategy` to :guilabel:`Least Packages`.

.. image:: removal/set-least-packages.png
   :align: center
   :alt: Set least packages removal strategy on the product category.

> Create a :ref:`delivery order <inventory/delivery/one-step>` for eighty kilograms of flour, by
> either going to the :menuselection:`Sales app` and create a new quotation, or from the delivery

Reviewer 3: ```suggestion
Create a :ref:`delivery order <inventory/delivery/one-step>` for eighty kilograms of flour,
either by going to the :menuselection:`Sales app` and creating a new quotation, or from the delivery
```

orders dashboard in :menuselection:`Inventory --> Operations --> Deliveries`.

On the delivery order, the :guilabel:`Quantity` field displays the amount automatically picked
according to the removal strategy. For more details about *where* the units were picked, select the

> :guilabel:`⦙≣ (bulleted list)` icon on the far right. Doing so opens the :guilabel:`Stock move`
> pop-up window that displays how the reserved items were picked according to the removal strategy.
> 
> In the :guilabel:`Stock move` window, the :guilabel:`Pick from` field details where the quantities

Reviewer 3: ```suggestion
:guilabel:`⦙≣ (bulleted list)` icon on the far right. Doing so opens the :guilabel:`Open: Stock move`
pop-up window that displays how the reserved items were picked according to the removal strategy.

In the :guilabel:`Open: Stock move` window, the :guilabel:`Pick from` field details where the quantities
```

to fulfill the :guilabel:`Demand` are picked from. Since the order demanded eighty kilograms, which
exceeds the quantity in the opened package of `54 kg`, an unopened package of `100 kg` is selected.

.. image:: removal/least-package.png
   :align: center
   :alt: Show which package was picked in the *Pick From* field.

Doing so allows the warehouse employees to consolidate the remaining `20 kg` of flour from the `100
kg` package with the `54 kg`, instead of picking the `54 kg` package, and picking the remaining `26

> kg` from another sealed package.

Reviewer 3: I would remove this. It seems outside the scope of the removal strategy.

Author: ```suggestion

```
